### PR TITLE
Add AWS volume plugin attach tests.

### PR DESCRIPTION
**Pull request by jsafrane**
@kubernetes/sig-storage 

This it a test, it does not really matter if it catches 1.3 train or the next one.

### CODE REVIEW

File: pkg/volume/aws_ebs/attacher_test.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	}
+}
+
+func newDetacher(testcase *testcase) *awsElasticBlockStoreDetacher {
+	return &awsElasticBlockStoreDetacher{
+		awsVolumes: testcase,
+	}
+}
+
+func createVSpec(name string, readOnly bool) *volume.Spec {
```
Comment (rootfs) [inline pkg/volume/aws_ebs/attacher_test.go]: createVolSpec for clarity.

File: pkg/volume/aws_ebs/attacher_test.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+type diskIsAttachedCall struct {
+	diskName, instanceID string
+	isAttached           bool
+	ret                  error
+}
+
+func (testcase *testcase) AttachDisk(diskName string, instanceID string, readOnly bool) (string, error) {
+	expected := &testcase.attach
+
+	if expected.diskName == "" && expected.instanceID == "" {
```
Comment (rootfs) [inline pkg/volume/aws_ebs/attacher_test.go]: && -> ||

File: pkg/volume/aws_ebs/attacher_test.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+	glog.V(4).Infof("AttachDisk call: %s, %s, %v, returning %q, %v", diskName, instanceID, readOnly, expected.retDeviceName, expected.ret)
+
+	return expected.retDeviceName, expected.ret
+}
+
+func (testcase *testcase) DetachDisk(diskName string, instanceID string) (string, error) {
+	expected := &testcase.detach
+
+	if expected.diskName == "" && expected.instanceID == "" {
```
Comment (rootfs) [inline pkg/volume/aws_ebs/attacher_test.go]: && -> ||

File: pkg/volume/aws_ebs/attacher_test.go [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+	glog.V(4).Infof("DetachDisk call: %s, %s, returning %q, %v", diskName, instanceID, expected.retDeviceName, expected.ret)
+
+	return expected.retDeviceName, expected.ret
+}
+
+func (testcase *testcase) DiskIsAttached(diskName, instanceID string) (bool, error) {
+	expected := &testcase.diskIsAttached
+
+	if expected.diskName == "" && expected.instanceID == "" {
```
Comment (rootfs) [inline pkg/volume/aws_ebs/attacher_test.go]: && -> ||

File: pkg/volume/aws_ebs/attacher_test.go (hunk 1 above)
Comment (jsafrane) [inline pkg/volume/aws_ebs/attacher_test.go]: Ok, but I'll change it also in `gce/attacher_test.go` to keep the test similar.

File: pkg/volume/aws_ebs/attacher_test.go (hunk 2 above)
Comment (jsafrane) [inline pkg/volume/aws_ebs/attacher_test.go]: I'd like to keep it as it is - I want this condition to match only uninitialized `testcase.attach` value, where **both** instanceID and diskName are empty.

**Comment by jsafrane** [comment]
`createVSpec` renamed to `createVolSpec`, PTAL

**Comment by rootfs** [comment]
LGTM

**Comment by jsafrane** [comment]
rebased

**Comment by saad-ali** [comment]
LGTM Thanks @jsafrane

**Pull request merged**